### PR TITLE
feat: add emit event to receiveDonation function

### LINKED ISSUE #138: [feat] Emit event when fund receives a donation
### This issue will be part of ODHack9.0, please apply via Onlydust app
- Emit an event when a fund contract receives a new donation in the ```receiveDonation``` method.
- The event should contain: address that is donating (caller address), the number of STRKs donated, the current STRKs balance (contract balance), and the fund contract address that is receiving the donation.
- Please read [contributors guide](https://github.com/web3wagers/gostarkme/blob/dev/CONTRIBUTORS_GUIDE.md) before asking for an issue

### Acceptance Criteria
- Code builds successfully.
- Event struct has all the attributes needed.

### LINKED ISSUE #139: [feat] Emit event when user withdraw fund's balance
### This issue will be part of ODHack9.0, please apply via Onlydust app
- Emit an event when a fund contract receives a new donation in the ```withdraw``` method.
- The event should contain: fund contract address, owner address, STRKs amount that is withdrawn.
- Please read [contributors guide](https://github.com/web3wagers/gostarkme/blob/dev/CONTRIBUTORS_GUIDE.md) before asking for an issue

### Acceptance Criteria
- Code builds successfully.
- Event struct has all the attributes needed.

### PR DESCRIPTION
**Pull request by NueloSE**
# Pull Request

- [x] Closes   #138
Open

- [ ] Added tests (if necessary)
- [x] Run tests
- [x] Run formatting
- [x] Commented the code

## Changes description

- Add emit event to receiveDonation function
- create a new struct
```cairo
    #[derive(Drop, starknet::Event)]
    pub struct DonationReceived {
        #[key]
        pub donor_address: ContractAddress,
        pub donated_strks: u256,
        pub current_balance: u256,
        pub fund_contract_address: ContractAddress,
    }
```

- Update Event
```
enum Event {
        DonationWithdraw: DonationWithdraw,
        NewVoteReceived: NewVoteReceived,
        DonationReceived: DonationReceived,
    }
    ```

## Current output

Give evidence like screenshots of what your job looks like.

## Time spent breakdown

< 5hrs

## Comments

formatted using `scarb fmt`

### CODE REVIEW

File: contracts/src/fund.cairo [hunk 1]
```diff
@@ -59,6 +60,15 @@ mod Fund {
         pub fund: ContractAddress,
         pub votes: u32
     }
+
+    #[derive(Drop, starknet::Event)]
+    pub struct DonationReceived {
+        #[key]
+        pub donor_address: ContractAddress,
```
Comment (EmmanuelAR) [inline contracts/src/fund.cairo:67]: typo: donator

File: contracts/src/fund.cairo [hunk 2]
```diff
@@ -187,7 +208,8 @@ mod Fund {
                 contract_address: starknet_contract_address
             };
             let balance = starknet_dispatcher.balance_of(get_contract_address());
-            //TODO: Calculate balance to deposit in owner address and in fund manager address (95% and 5%), also transfer the amount to fund manager address.
+            //TODO: Calculate balance to deposit in owner address and in fund manager address (95%
```
Comment (EmmanuelAR) [inline contracts/src/fund.cairo:211]: Leave original, do not separate it

**Review by EmmanuelAR — CHANGES_REQUESTED** [review]
Exc work @NueloSE just some changes!

**Review by EmmanuelAR — APPROVED** [review]
LGTM

File: contracts/src/fund.cairo [hunk 3]
```diff
@@ -160,6 +170,17 @@ mod Fund {
             if self.current_goal_state.read() >= self.goal.read() {
                 self.state.write(FundStates::CLOSED);
             }
+
+            // Emit receiveDonation event
+            self
+                .emit(
+                    DonationReceived {
+                        donor_address: get_caller_address(),
```
Comment (EmmanuelAR) [inline contracts/src/fund.cairo:178]: Also change the event, please run the scarb build and fmt

**Review by EmmanuelAR — CHANGES_REQUESTED** [review]

**Comment by NueloSE** [comment]
@EmmanuelAR  fixed the typo now

File: contracts/src/fund.cairo [hunk 4]
```diff
@@ -59,6 +60,15 @@ mod Fund {
         pub fund: ContractAddress,
         pub votes: u32
     }
+
+    #[derive(Drop, starknet::Event)]
+    pub struct DonationReceived {
+        #[key]
+        pub current_balance: u256,
```
Comment (EmmanuelAR) [inline contracts/src/fund.cairo:67]: the key will be the donator address not the current balance

**Review by EmmanuelAR — CHANGES_REQUESTED** [review]

**Pull request merged**